routing/payment_lifecycle.go

```diff
@@ -782,10 +782,13 @@ func (p *shardHandler) handleSendError(attempt *channeldb.HTLCAttemptInfo,
 		if err := p.router.cfg.Control.Fail(
 			p.identifier, *reason); err != nil {
 
-			return err
+			log.Errorf("unable to report failure to control "+
+				"tower: %v", err)
+
+			return &internalErrorReason
```

Comment:
Similar to ReportPaymentFail below, we just count this as an internal failure reason. This lets the method always return *channeldb.FailureReason.

The downside is that we lose this error context from the PoV, of the caller.

Reply:
I'd prefer the `internalErrorReason`. To me errors come from other packages/services do seem to be "internal" when we are viewing it as the user of the `routing`. It's probably better (haven't thought through it) though we could wrap the error inside the internal error and returns it back to the caller.